### DISCUSSION TITLE
Hello github assistant, I want to know why it is so difficult to use the new search code, and why github deleted the old code search, will the old code search be restored?

### DISCUSSION THREAD

**Discussion by cobavcova** [community/community#114201] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello github assistant, I want to know why it is so difficult to use the new search code, and why github deleted the old code search, will the old code search be restored?

**Comment by cobavcova** [comment]
please reply to my feedback

**Comment by xiekunwhy** [comment]
In my memory, github code search results in about 2 years ago was much better than now.

**Reply by cobavcova** [reply]
I'm more interested in the old search code, in fact I'm not the only one who needs the old search code, everyone really needs how useful the old search code is for learning

**Comment by mike-clark-8192** [comment]
It might help if you could add some specific details to your request.
For example: make a prioritized list the features of the old code search that you found useful (that are lacking in the new code search).
For each, specify some reasons why they were useful to you (use-cases).

**Reply by cobavcova** [reply]
I'm more interested in the old search code, in fact I'm not the only one who needs the old search code, everyone really needs how useful the old search code is for learning

**Reply by LiteBrite82** [reply]
Hi @cobavcova, sorry to hear about your difficulties with the code search; for sure, there were some big changes a while ago. 

Can you explain your use case in a little more detail? 

For example: "I'm an instructor at a coding bootcamp, and we use the code search function to check for plagiarism in student submissions. I was previously able to search for {specific search terms and filters here} and receive {example of search return}. Now, with the new search, If I use the same terms and filters, I receive {new search terms example}." "I am not receiving { desired outcome } any more"

**Reply by cobavcova** [reply]
hello coding instructor, I just want to ask when will search codes by date or old search codes be returned? or there is a way to search for a code that is current or that has not appeared in the past year

**Reply by cobavcova** [reply]
Sorry in advance, I only want to use code based on date and rows reaching 100, can this be granted?

**Reply by LiteBrite82** [reply]
Are you looking for specific types of code snippets like: search all of GitHub for files that have this line of code: 
```
require('dotenv').config();
const inquirer = require('inquirer');
```

In {x} time frame?

Or, are you looking for full repos that include code that pertains to a specific tech stack.

**Reply by LiteBrite82** [reply]
In the advanced search options you can filter for repositories "created on the dates" and "of this size" though I don't believe there is a filter for rows of code.

**Reply by cobavcova** [reply]
Yes, like the old search, is it possible?

**Reply by cobavcova** [reply]
I hope it all comes back again

**Reply by LiteBrite82** [reply]
without a more detailed understanding of your overall goal, its hard for me to tell if the functionality you are looking for is available. From my experience, having used the code search engine since 2020, the new search engine still preforms much the same as it did; though the search process - how to search - became much more detailed. It is more of a UI thing, from my perspective. 

Though that is through the lens of my use cases and is not representative of your experience. 

What I can definitively say is that going through the [documentation](https://docs.github.com/en/search-github/github-code-search/about-github-code-search) helped me better understand how to search for code. I used to just copy / paste whole lines of code into the search bar and receive thousands of returns that I would then have to browse through. Now the engine is geared towards specificity, accuracy and, efficiency. 

I don't want to come across as dismissive by just providing a link to the documentation; that doesn't help us build a healthier community. Though I could be more helpful with a better understanding of your specific use case.

**Reply by cobavcova** [reply]
yes, because everything I searched for last year's results, there were no new indexes, and at the beginning of 2023, Github also launched 2 features, namely the old search feature and the new search feature, and now why has the old search been deleted, even though the new search is more precise and accurate but it doesn't have a new date but instead shows old results, so far the bad experience I have learned on GitHub is that the code search is not perfect, and we searched for 100 chapters, now GitHub only displays 5 chapters, thank you, sorry in advance, best regards

**Reply by xiekunwhy** [reply]
@cobavcova this is exactly what I want to say, are you sure "the new search is more precise and accurate" ? I think the old search is more precise and accurate, the new search is cut my sentence as unrelated segments and give unrelated results.

**Reply by xiekunwhy** [reply]
why off-topic? has not been resolved.

**Reply by xiekunwhy** [reply]
@LiteBrite82 the request is so clearly now, we just want the old code search feature back or at least give us an option to choose old search feature or the new search feature like that in the beginning of 2023. If there is a time machine, I will go back and show you many difference results between the old search feature and the new search. We can not show more details now just because the old code search feature was gone.

**Reply by cobavcova** [reply]
@xiekunwhy Thank you for representing my feelings, sorry if there is a mistake like Happy Eid Mubarak im Muslim

**Reply by cobavcova** [reply]
It's been a year since Github removed the old code search feature, we ask the admins to please relaunch the old code search system or create 2 features

**Reply by cobavcova** [reply]
![image](https://github.com/community/community/assets/110930268/e15d6674-e93f-484a-be43-f437be61efe6)


this is an example of a maximum long search

**Reply by cobavcova** [reply]
please grant our request

**Reply by cobavcova** [reply]
![image](https://github.com/community/community/assets/110930268/914cb286-fe10-4426-aab8-8d3ea382384d)

please make a feature like this again

**Reply by xiekunwhy** [reply]
@LiteBrite82 please grant our request, and users following this discussions https://github.com/orgs/community/discussions/52932 as well.

highly agree with "even Microsoft Windows 95 had a better search engine almost 30 years ago."

![image](https://github.com/community/community/assets/17738229/372783c7-d68e-4c02-aec3-5dbd30b9315a)

**Reply by cobavcova** [reply]
@LiteBrite82 no response?

**Comment by aryejfa** [comment]
Response #10135

**Reply by cobavcova** [reply]
hello github how will this be resolved?

**Comment by cobavcova** [comment]
![image](https://github.com/community/community/assets/110930268/6fd7625a-e584-4914-8141-8611377a30c7)

We've been waiting for a year, sir


![image](https://github.com/community/community/assets/110930268/1f58bac8-65e2-4730-a78e-1859f1f8be74)


how many people lose by this

**Reply by xiekunwhy** [reply]
@cobavcova don't forget to \@ @LiteBrite82 , hahahaha^_^

**Reply by LiteBrite82** [reply]
👀 looking into it

**Reply by cobavcova** [reply]
is that your only response?

**Reply by cobavcova** [reply]
@LiteBrite82 So GitHub won't change anything, right?

**Reply by cobavcova** [reply]
@LiteBrite82 any update sir?

**Reply by LiteBrite82** [reply, marked as answer]
I found some previous answers to this question in another post:

https://github.com/orgs/community/discussions/54546

**Reply by xiekunwhy** [reply]
@LiteBrite82 so the elder version of code search never come back again, right? For me, I don't really understand and don't really care about the explanation, just tell you the experience and hope that the old code search feature can come back as I found that github code search was help me a lot in those years(before 2023 if I remember correct), but hard to get helpful results since we can't choose old code search feature.

**Reply by cobavcova** [reply]
@LiteBrite82 So the old code search engine won't be updated, right?

**Reply by cobavcova** [reply]
@LiteBrite82 response please

**Comment by ettaboyle** [comment]
@cobavcova Please note, **contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment.** Please review our [Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct) for more information.

I know that `LiteBrite82` has been responsive and helping here, but please remember, this community is managed by GitHub employees, but it is not a support forum and **there is no SLA or guarantee each post will get a response from a community manager or product leader.**

If your discussion here (or any others you share) provided product feedback or a feature request we encourage you to review the "Making Suggestions" [guidelines](https://github.com/community/community#making-suggestions) in the repo readme including: "Once you kick off a discussion, the GitHub product team will evaluate the feedback but will not be able to respond to every submission."

Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.

Any repeat and future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.

**Reply by cobavcova** [reply]
hello, sorry, I don't know, I just want to ask for confirmation, please understand, because I'm not the only one having difficulty with this problem, thank you, best regards
